### ISSUE TITLE
Failing with installed extensions

### ISSUE THREAD

**Issue by fdw** [pgautoupgrade/docker-pgautoupgrade#17]
So far, I've used the official Docker image `postgres:15`, but I've also enabled some built-in extensions (f.e. `cube`). Now I wanted to upgrade to v16 using your nice project, but unfortunately it fails with `pg_dump: error: query failed! ERROR: could not access file "$libdir/cube": No such file or directory"

I suspect I need to create the extension before the script kicks in, but I haven't found a way to do that. Or is the extension not contained at all?

**Comment by justinclift** [comment]
Oh, now that's an aspect of things I hadn't considered.

Um, give me a few days to look into it?  I hadn't even thought of the whole PG extensions side of things.

**Comment by fdw** [comment]
No worries. It's even fine if you say that extensions are out of scope, the container is still very helpful 🙂

Maybe it's possible to specify extensions via a env var? Or a script will be executed if it's mounted in the correct place? (That would even be useful for non-default extensions.)

**Comment by justinclift** [comment]
Not sure yet, I haven't investigated to see how the official Docker PG containers handle extensions.  Am wanting to check that first before making promises, plans, etc. :smile:

**Comment by justinclift** [comment]
@fdw Just added the common PostgreSQL extensions to the pgautoupgrade Docker images.

If you're up for trying things again, then (in theory) it should work for you now. :smile:

**Comment by fdw** [comment]
Sorry, I had to move away from Postgres for different reasons :/

**Comment by justinclift** [comment]
No worries at all.  It's been ~3 months since you reported the problem, so I don't blame you at all. :wink:

**Comment by justinclift** [comment]
I'll close this issue then.  PG extensions should (in theory) be working for people now.

**Comment by spwoodcock** [comment]
The standard bundled extensions work fine, but if you need a custom extension extra work is required.

### Issue

I was trying to migrate a db that included PostGIS.

I managed to get `tianon/docker-postgres-upgrade` to work well from a **Debian** based database container: https://github.com/tianon/docker-postgres-upgrade/discussions/101

However, I needed to upgrade an **Alpine** based database container, and there are no Alpine-based images available in that repo.

Upgrading an Alpine-based container using a Debian-based pgupgrade causes a well documented collation version issue.
**Warning: upgrading an Alpine db with a Debian pgupgrade may corrupt your data!**

`pgautoupgrade/docker-pgautoupgrade` to the rescue! Thank you for your work on this tool 😄 

### Adding PostGIS to pgautoupgrade

I assume a similar process could be followed to install any additional extension you may need.

Dockerfile:

```dockerfile
FROM docker.io/postgis/postgis:14-3.4-alpine as pg14
FROM docker.io/postgis/postgis:15-3.4-alpine as pg15
FROM docker.io/postgis/postgis:16-3.4-alpine as pg16
FROM docker.io/pgautoupgrade/pgautoupgrade:16-alpine as pgautoupgrade

# Original entrypoint copied overwritten in /usr/local/bin, so copy elsewhere
RUN cp /usr/local/bin/docker-entrypoint.sh /docker-entrypoint.sh

# Remove libs for Postgres < v14
RUN rm -rf \
    /usr/local-pg9.5 \
    /usr/local-pg9.6 \
    /usr/local-pg10 \
    /usr/local-pg11 \
    /usr/local-pg12 \
    /usr/local-pg13

# Copy in PostGIS libs / bins
COPY --from=pg16 /_pgis*.* /
COPY --from=pg16 /usr/lib /usr/lib

# Copy extensions for postgresql 16
COPY --from=pg16 /usr/local /usr/local
# Copy all extensions for postgresql 15
COPY --from=pg15 /usr/local/lib/postgresql /usr/local-pg15/lib/postgresql
# Copy all extensions for postgresql 14
COPY --from=pg14 /usr/local/lib/postgresql /usr/local-pg14/lib/postgresql

# Squash image to reduce size
FROM scratch
COPY --from=pgautoupgrade / /
ENV \
    PGTARGET=16 \
    PGDATA=/var/lib/postgresql/data
WORKDIR /var/lib/postgresql
ENTRYPOINT ["/docker-entrypoint.sh"]
CMD ["postgres"]
```

This dockerfile:
- Strips out any versions of postgres prior to v15.
- Bundles the PostGIS extension built for each version of Postgres (14, 15, 16).
- Flattens the image to reduce size to `604MB`.

How to use this:
- Build the image.
- Use the image in place of the standard `pgautoupgrade/pgautoupgrade` one.
- Upgrade from Postgres v14 --> v16 or v15 --> v16.

**Comment by justinclift** [comment]
Cool, that looks like it'll be useful for people! :grin:

Any interest in throwing it in the wiki as a more official reference thing of some sort? :smile:

**Comment by spwoodcock** [comment]
Very happy to do that!

Of there a preferred place to do that?

It could go in the README, but probably belongs in a docs or wiki page as you say.

Either I could:
- Configure an mkdocs Github pages docs for this repo.
- Create a blog on dev.to and reference that. 
- Another suggestion 😃

**Comment by justinclift** [comment]
Ahhh.  Sorry, I didn't consider that the wiki tab for this repo might not have been visible to people.

I've just gone and created an initial front page for the wiki, so please feel free to throw useful stuff in there:

https://github.com/pgautoupgrade/docker-pgautoupgrade/wiki

**Comment by justinclift** [comment]
Awesome!  Just added a link to that from the front page of the wiki too, so it's easier for people to find.

We should probably add some kind of pointer towards the wiki itself too, now that I think about it. :smile:

**Comment by spwoodcock** [comment]
I hope you don't mind that I also took the liberty of adding an extra page about automating db upgrades:
https://github.com/pgautoupgrade/docker-pgautoupgrade/wiki

I had to do this myself, so thought it best to document 😀

**Comment by justinclift** [comment]
> I hope you don't mind

No worries at all.  The more helpful/useful stuff there for people, the better. :smile:

**Comment by FireHelmet** [comment]
Hello @spwoodcock ,

Thank you for the dockerfile based on your example. I'm struggling to do the same but with TimescaleDB extension.

May I request your help ?

```Dockerfile
FROM docker.io/timescale/timescaledb:2.15.1-pg14 as pg14
FROM docker.io/timescale/timescaledb:2.15.1-pg15 as pg15
FROM docker.io/timescale/timescaledb:2.15.1-pg16 as pg16
FROM docker.io/pgautoupgrade/pgautoupgrade:16-alpine as pgautoupgrade

# Original entrypoint overwritten in /usr/local/bin, so copy elsewhere
RUN cp /usr/local/bin/docker-entrypoint.sh /docker-entrypoint.sh

# Remove libs for Postgres < v14
RUN rm -rf \
    /usr/local-pg9.5 \
    /usr/local-pg9.6 \
    /usr/local-pg10 \
    /usr/local-pg11 \
    /usr/local-pg12 \
    /usr/local-pg13

COPY --from=pg16 /usr/lib /usr/lib

# Copy extensions for postgresql 16
COPY --from=pg16 /usr/local /usr/local
# Copy all extensions for postgresql 15
COPY --from=pg15 /usr/local/lib/postgresql /usr/local-pg15/lib/postgresql
# Copy all extensions for postgresql 14
COPY --from=pg14 /usr/local/lib/postgresql /usr/local-pg14/lib/postgresql

# Squash image to reduce size
FROM scratch
COPY --from=pgautoupgrade / /
ENV \
    PGTARGET=16 \
    PGDATA=/var/lib/postgresql/data
WORKDIR /var/lib/postgresql
ENTRYPOINT ["/docker-entrypoint.sh"]
CMD ["postgres"]
```

Then I execute as:

```Docker
docker run --name pgauto -it \
	--mount type=bind,source=/SOURCE/_data,target=/var/lib/postgresql/data \
	-e POSTGRES_PASSWORD=MY_PASSWORD \
	-e PGAUTO_ONESHOT=yes \
	pgautoupgrade:pg16-2.15.1
```

But I get only this output:

```bash
PostgreSQL Database directory appears to contain a database; Skipping initialization

************************************
PostgreSQL data directory: /var/lib/postgresql/data
************************************
*****************************************************************************************************
'One shot' automatic upgrade was requested, so exiting now rather than starting the PostgreSQL server
*****************************************************************************************************
```

Thanks in advance !

**Comment by FireHelmet** [comment]
Hello @spwoodcock 

I executed again the command in a fresh environment and it has worked partially but more than before :).
Nevertheless, I'm blocked by this new error, it seems a "collation" issue, right ?

```bash
creating subdirectories ... ok
selecting dynamic shared memory implementation ... posix
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default time zone ... UTC
creating configuration files ... ok
running bootstrap script ... ok
performing post-bootstrap initialization ... 2024-08-14 10:23:39.099 UTC [54] FATAL:  could not open collator for locale "und": U_FILE_ACCESS_ERROR
2024-08-14 10:23:39.099 UTC [54] STATEMENT:  UPDATE pg_collation SET collversion = pg_collation_actual_version(oid) WHERE collname = 'unicode';

child process exited with exit code 1
initdb: removing contents of data directory "/var/lib/postgresql/data/new"
```

Is it the error you indicated in the wiki here ?
https://github.com/pgautoupgrade/docker-pgautoupgrade/wiki/Including-Extensions-(PostGIS)

My OS in the container where the database is currently running is based on Alpine, also. But 3.18 and not 3.20.
```bash
0cd198760ac5:/# more /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.18.6
PRETTY_NAME="Alpine Linux v3.18"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://gitlab.alpinelinux.org/alpine/aports/-/issues"
```

Thank you for the support !

**Comment by spwoodcock** [comment]
Actually I don't think it's the error from the wiki!

I am assuming you upgraded from Alpine to Alpine, so it should be fine.

It looks more likely to me that the issue is a missing locale on your system, or something related.

Sorry I'm no postgres wizard 😅

**Comment by zN3utr4l** [comment]
Same error here. @FireHelmet 

https://forums.freebsd.org/threads/postgresql12-server-upgrade-fails.82677/
https://postgrespro.com/list/thread-id/2329582

These 2 forums seemed to have found the solution but it doesn't work even adding this part.

```Dockerfile
# install ICU + chrpath rpath
RUN apk add --no-cache icu-libs icu-data-full chrpath

# patch rpath ICU
RUN for lib in /usr/lib/libicu*.so*; do \
      chrpath -r /usr/lib "$lib" || true; \
    done
```

```Dockerfile
FROM timescale/timescaledb:2.19.3-pg14 AS pg14
FROM timescale/timescaledb:2.19.3-pg15 AS pg15
FROM timescale/timescaledb:2.19.3-pg16 AS pg16
FROM timescale/timescaledb:2.19.3-pg17 AS pg17
FROM pgautoupgrade/pgautoupgrade:17-alpine3.22 AS pgautoupgrade

RUN cp /usr/local/bin/docker-entrypoint.sh /docker-entrypoint.sh

RUN rm -rf /usr/local-pg9* /usr/local-pg1[0-3]

COPY --from=pg17 /usr/lib /usr/lib
COPY --from=pg17 /usr/local /usr/local
COPY --from=pg14 /usr/local/lib/postgresql /usr/local-pg14/lib/postgresql
COPY --from=pg15 /usr/local/lib/postgresql /usr/local-pg15/lib/postgresql
COPY --from=pg16 /usr/local/lib/postgresql /usr/local-pg16/lib/postgresql

FROM scratch

COPY --from=pgautoupgrade / /

ENV PGTARGET=17 PGDATA=/var/lib/postgresql/data

WORKDIR /var/lib/postgresql

ENTRYPOINT ["/docker-entrypoint.sh"]

CMD ["postgres"]

# *** RUN ***
# docker run --rm -v timescale_data:/from:ro -v timescale_data_backup:/to alpine sh -c "cd /from && tar cf - . | tar xf - -C /to"

# docker build -f .\TimeScaleDB\Dockerfile.pgautoupgrade -t my/pgautoupgrade:pg14-to-pg17 .
# docker run --rm -it --mount type=volume,source=timescale_data,target=/var/lib/postgresql/data -e PGAUTO_ONESHOT=yes my/pgautoupgrade:pg14-to-pg17
```

Also there is this error: **mv: can't rename '/var/lib/postgresql/data/old': Invalid argument**

```Terminal
PostgreSQL Database directory appears to contain a database; Skipping initialization

************************************
PostgreSQL data directory: /var/lib/postgresql/data
************************************
Creating upgrade lock file at /var/lib/postgresql/data/upgrade_in_progress.lock
*******************************************************************************************
Performing PG upgrade on version 14 database files.  Upgrading to version 17
*******************************************************************************************
----------------------------------------------------------------------
Checking for left over artifacts from a failed previous autoupgrade...
----------------------------------------------------------------------
-------------------------------------------------------------------------------
No artifacts found from a failed previous autoupgrade.  Continuing the process.
-------------------------------------------------------------------------------
---------------------------------------
Creating OLD temporary directory /var/lib/postgresql/data/old
---------------------------------------
--------------------------------------------
Creating OLD temporary directory is complete
--------------------------------------------
-------------------------------------------------------
Moving existing data files into OLD temporary directory
-------------------------------------------------------
'/var/lib/postgresql/data/PG_VERSION' -> '/var/lib/postgresql/data/old/PG_VERSION'
'/var/lib/postgresql/data/base' -> '/var/lib/postgresql/data/old/base'
'/var/lib/postgresql/data/current_logfiles' -> '/var/lib/postgresql/data/old/current_logfiles'
'/var/lib/postgresql/data/global' -> '/var/lib/postgresql/data/old/global'
'/var/lib/postgresql/data/log' -> '/var/lib/postgresql/data/old/log'
mv: can't rename '/var/lib/postgresql/data/old': Invalid argument
'/var/lib/postgresql/data/old' -> '/var/lib/postgresql/data/old/old'
'/var/lib/postgresql/data/pg_commit_ts' -> '/var/lib/postgresql/data/old/pg_commit_ts'
'/var/lib/postgresql/data/pg_dynshmem' -> '/var/lib/postgresql/data/old/pg_dynshmem'
'/var/lib/postgresql/data/pg_hba.conf' -> '/var/lib/postgresql/data/old/pg_hba.conf'
'/var/lib/postgresql/data/pg_ident.conf' -> '/var/lib/postgresql/data/old/pg_ident.conf'
'/var/lib/postgresql/data/pg_logical' -> '/var/lib/postgresql/data/old/pg_logical'
'/var/lib/postgresql/data/pg_multixact' -> '/var/lib/postgresql/data/old/pg_multixact'
'/var/lib/postgresql/data/pg_notify' -> '/var/lib/postgresql/data/old/pg_notify'
'/var/lib/postgresql/data/pg_replslot' -> '/var/lib/postgresql/data/old/pg_replslot'
'/var/lib/postgresql/data/pg_serial' -> '/var/lib/postgresql/data/old/pg_serial'
'/var/lib/postgresql/data/pg_snapshots' -> '/var/lib/postgresql/data/old/pg_snapshots'
'/var/lib/postgresql/data/pg_stat' -> '/var/lib/postgresql/data/old/pg_stat'
'/var/lib/postgresql/data/pg_stat_tmp' -> '/var/lib/postgresql/data/old/pg_stat_tmp'
'/var/lib/postgresql/data/pg_subtrans' -> '/var/lib/postgresql/data/old/pg_subtrans'
'/var/lib/postgresql/data/pg_tblspc' -> '/var/lib/postgresql/data/old/pg_tblspc'
'/var/lib/postgresql/data/pg_twophase' -> '/var/lib/postgresql/data/old/pg_twophase'
'/var/lib/postgresql/data/pg_wal' -> '/var/lib/postgresql/data/old/pg_wal'
'/var/lib/postgresql/data/pg_xact' -> '/var/lib/postgresql/data/old/pg_xact'
'/var/lib/postgresql/data/postgresql.auto.conf' -> '/var/lib/postgresql/data/old/postgresql.auto.conf'
'/var/lib/postgresql/data/postgresql.conf' -> '/var/lib/postgresql/data/old/postgresql.conf'
'/var/lib/postgresql/data/postmaster.opts' -> '/var/lib/postgresql/data/old/postmaster.opts'
'/var/lib/postgresql/data/postmaster.pid' -> '/var/lib/postgresql/data/old/postmaster.pid'
'/var/lib/postgresql/data/upgrade_in_progress.lock' -> '/var/lib/postgresql/data/old/upgrade_in_progress.lock'
-------------------------------------------------------------------
Moving existing data files into OLD temporary directory is complete
-------------------------------------------------------------------
---------------------------------------
Creating NEW temporary directory /var/lib/postgresql/data/new
---------------------------------------
--------------------------------------------
Creating NEW temporary directory is complete
--------------------------------------------
-----------------------------------------------------
Changing permissions of temporary directories to 0700
-----------------------------------------------------
---------------------------------------------------------
Changing permissions of temporary directories is complete
---------------------------------------------------------
-------------------------------------------------
Remove postmaster.pid file from PG data directory
-------------------------------------------------
------------------------------------
Determining our own initdb arguments
------------------------------------
2025-06-20 09:52:17.466 UTC [42] LOG:  database system was interrupted; last known up at 2025-06-20 08:24:25 UTC
2025-06-20 09:52:18.359 UTC [42] LOG:  database system was not properly shut down; automatic recovery in progress
2025-06-20 09:52:18.362 UTC [42] LOG:  redo starts at 25/871EFAD8
2025-06-20 09:52:18.365 UTC [42] LOG:  invalid record length at 25/871FDE08: wanted 24, got 0
2025-06-20 09:52:18.365 UTC [42] LOG:  redo done at 25/871FDDD0 system usage: CPU: user: 0.00 s, system: 0.00 s, elapsed: 0.00 s
---------------------------------------------------------------
The initdb arguments we determined are: --locale=en_US.utf8 --lc-collate=en_US.utf8 --lc-ctype=en_US.utf8 --encoding=UTF8
---------------------------------------------------------------
--------------------------------------------------------------------------------------------------------------------
Old database using collation settings: '--locale=en_US.utf8 --lc-collate=en_US.utf8 --lc-ctype=en_US.utf8 --encoding=UTF8'.  Initialising new database with those settings too
--------------------------------------------------------------------------------------------------------------------
Initialising PostgreSQL 17 data directory
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.utf8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /var/lib/postgresql/data/new ... ok
creating subdirectories ... ok
selecting dynamic shared memory implementation ... posix
selecting default "max_connections" ... 100
selecting default "shared_buffers" ... 128MB
selecting default time zone ... UTC
creating configuration files ... ok
running bootstrap script ... ok
performing post-bootstrap initialization ... 2025-06-20 09:52:19.116 UTC [61] FATAL:  could not open collator for locale "und": U_FILE_ACCESS_ERROR
2025-06-20 09:52:19.116 UTC [61] STATEMENT:  UPDATE pg_collation SET collversion = pg_collation_actual_version(oid) WHERE collname = 'unicode';

child process exited with exit code 1
initdb: removing contents of data directory "/var/lib/postgresql/data/new"
```

**Comment by zN3utr4l** [comment]
Thanks @FireHelmet but nothing to do, it gives me this error, I searched and it seems that in the "OLD" volume there are files from an inaccurate shutdown, so I tried again to activate my timescale container and shut it down with the stop button of the docker container, redo the procedure but nothing. (gosu was not available so I installed su-exec)

```Docekrfile
FROM timescale/timescaledb:2.19.3-pg14 AS timescaledb14-base

RUN apk update
RUN apk add --no-cache musl-locales postgresql14 su-exec

FROM timescale/timescaledb:2.19.3-pg15

RUN set -eux; \
	apk update; \
	apk add --no-cache musl-locales postgresql14 su-exec

COPY --from=timescaledb14-base /usr/local/lib/postgresql /usr/lib/postgresql14/

ENV POSTGRES_USER="........."
ENV POSTGRES_DB="................"
ENV POSTGRES_PASSWORD="................"
ENV POSTGRES_HOST_AUTH_METHOD=scram-sha-256
ENV EXTENSIONS=""

ENV PGBINOLD=/usr/libexec/postgresql14
ENV PGBINNEW=/usr/local/bin

ENV PGDATAOLD=/var/lib/postgresql/14/data
ENV PGDATANEW=/var/lib/postgresql/15/data

RUN set -eux; \
	mkdir -p "${PGDATAOLD}" "${PGDATANEW}"; \
	chown -R postgres:postgres /var/lib/postgresql

WORKDIR /var/lib/postgresql

COPY entrypoint.pgupgrade.sh /usr/local/bin/

RUN chmod 775 /usr/local/bin/entrypoint.pgupgrade.sh

ENTRYPOINT ["/usr/local/bin/entrypoint.pgupgrade.sh"]

# recommended: --link
CMD ["pg_upgrade", "-U", "${POSTGRES_USER}" ]
```

```bash
#!/bin/bash
set -e

if test "$#" -eq 0 -o "${1:0:1}" = '-'; then
	set -- pg_upgrade "$@" -U ${POSTGRES_USER}
fi

OPT=$1

case "${OPT}" in
pg_upgrade)
	if test "$(id -u)" = '0'; then
		mkdir -p "${PGDATAOLD}" "${PGDATANEW}"
		chmod 700 "${PGDATAOLD}" "${PGDATANEW}"
		chown postgres .
		chown -R postgres "${PGDATAOLD}" "${PGDATANEW}"
		exec su-exec postgres "${BASH_SOURCE}" "$@" -U ${POSTGRES_USER}
	fi

	# Address https://github.com/tianon/docker-postgres-upgrade/issues/100
	if test ! -s "${PGDATANEW}/PG_VERSION"; then
		PGDATA="${PGDATANEW}" eval "initdb -U ${POSTGRES_USER} --auth-host=${POSTGRES_HOST_AUTH_METHOD} --pwfile=<(printf '%s\n' '${POSTGRES_PASSWORD}') ${POSTGRES_INITDB_ARGS}"
	fi
	;;
esac

# Address https://github.com/tianon/docker-postgres-upgrade/issues/1
echo "host all all all ${POSTGRES_HOST_AUTH_METHOD}" >> ${PGDATANEW}/pg_hba.conf

echo "Perform upgrade"
eval "$@"
echo "Upgrade done"

if test -n "${EXTENSIONS}"; then
	echo "Update extensions: ${EXTENSIONS}"
	for EXT in ${EXTENSIONS}
	do
cat << EOF >>/tmp/update_extensions.sql
\\connect template1
ALTER EXTENSION "${EXT}" UPDATE;
\\connect postgres
ALTER EXTENSION "${EXT}" UPDATE;
\\connect ${POSTGRES_DB}
ALTER EXTENSION "${EXT}" UPDATE;
EOF
	done
	psql -U ${POSTGRES_USER} -d ${POSTGRES_DB} -a -f /tmp/update_extensions.sql
	echo "Update extensions done"
else
	echo "No extensions to update"
fi
```

```terminal
docker build -f Dockerfile.pgupgrade.14-to-15 -t pgupgrade:14-to-15 . ; docker run --rm -it -v timescale_data:/var/lib/postgresql/14/data -v timescale_data_15:/var/lib/postgresql/15/data pgupgrade:14-to-15


The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.utf8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /var/lib/postgresql/15/data ... ok
creating subdirectories ... ok
selecting dynamic shared memory implementation ... posix
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default time zone ... UTC
creating configuration files ... ok
running bootstrap script ... ok
performing post-bootstrap initialization ... ok
syncing data to disk ... ok

initdb: warning: enabling "trust" authentication for local connections
initdb: hint: You can change this by editing pg_hba.conf or using the option -A, or --auth-local and --auth-host, the next time you run initdb.

Success. You can now start the database server using:

    pg_ctl -D /var/lib/postgresql/15/data -l logfile start

Perform upgrade

There seems to be a postmaster servicing the old cluster.
Please shutdown that postmaster and try again.
Failure, exiting
```

**Comment by spwoodcock** [comment]
Thanks for all of the valuable info @zN3utr4l @FireHelmet!

Am I right in saying that you managed to successfully upgrade with the info provided [in the wiki article](https://github.com/pgautoupgrade/docker-pgautoupgrade/wiki/Including-Extensions-(PostGIS))? But the problem you are facing is a different one about socket errors?

Is there anything you feel needs to be added to the wiki to help clarify the upgrade process with extensions? 🙏

**Comment by zN3utr4l** [comment]
No, I failed to upgrade with docker-pgautoupgrade as you can see in issue #127, I solved it by calling pg_upgrade directly. https://github.com/timescale/timescaledb/issues/8286#issuecomment-3017963613
